R/consecutive_stopifnot_linter.R

```diff
@@ -1,48 +0,0 @@
-#' Force consecutive calls to stopifnot into just one when possible
-#'
-#' [stopifnot()] accepts any number of tests, so sequences like
-#'   `stopifnot(x); stopifnot(y)` are redundant.
-#'
-#' @examples
-#' # will produce lints
-#' lint(
-#'   text = "stopifnot(x); stopifnot(y)",
-#'   linters = consecutive_stopifnot_linter()
-#' )
-#'
-#' # okay
-#' lint(
-#'   text = "stopifnot(x, y)",
-#'   linters = consecutive_stopifnot_linter()
-#' )
-#'
-#' @evalRd rd_tags("consecutive_stopifnot_linter")
-#' @seealso [linters] for a complete list of linters available in lintr.
-#' @export
-consecutive_stopifnot_linter <- function() {
-  # match on the expr, not the SYMBOL_FUNCTION_CALL, to ensure
```

Comment:
NB: this comment intentionally removed as the logic now lints on cases like:

```r
assert_that(x)
assertthat::assert_that(y)
```

Technically it would also lint on

```r
SomePkg::stopifnot(x)
base::stopifnot(y)
```

But that seems unlikely in practice. We could extend the logic to something like "lint if the namespaces are equal or both absent, or if only present once, but not if both present and not equal" but that seems like overkill.